torch/_inductor/triton_helpers.py

```diff
@@ -92,22 +93,42 @@ def max_with_index(value, index, dim):
 
 
 @triton.jit
-def welford_reduce(value, mean, m2, weight):
-    delta = value - mean
-    new_weight = weight + 1
-    new_mean = mean + delta / new_weight
-    return (
-        new_mean,
-        m2 + delta * (value - new_mean),
-        new_weight,
-    )
+def div_approx(a, b):
+    return a * tl.math.rcp_rn(b)
```

Comment:
`rcp.rn` is just a different rounding. If you want to use the fast one you should probably use `approx`. Otherwise, is `a * (1/b)` really faster than `a/b`? I've been looking around and it seems that most implementations of `a/b` do it via multiplication + inversion.

Reply:
> rcp.rn is just a different rounding. If you want to use the fast one you should probably use approx.

`div.approx` isn't directly exposed in the triton language, however if you look at the ptx manual it states
> `div.approx.f32` implements a fast approximation to divide, computed as d = a * (1/b).
so this is equivalent.

There is also an `rcp.approx` which might speed things up further but would need inline assembly to use it.

I couldn't find latency tables for a current gen GPU but this paper gives ~2x latency improvement on a TITAN RTX. (Table IV at the bottom)
https://arxiv.org/pdf/1905.08778.pdf

Reply:
I found a source for A100, that claims `div.rn.f32` is 525 cycles/inst vs 198 for `rcp.rn.f32`.

https://arxiv.org/pdf/2208.11174.pdf (Table V on page 7)

Reply:
That table is great! Look at that `rcp.approx.f32` at 23 cycles. I wonder whether that'd be usable at all...